### PR TITLE
fix mismatch of analyze return code and pass/fail result

### LINKED ISSUE #28: Speedo analyze result pass/fails are mismatched
Screen Shot 2019-04-15 at 1 28 37 PM

Noticed this oddity when we were testing Speedo in Jenkins and was also reported as 1) in https://github.com/saucelabs/speedo/issues/23

As you can see in this screenshot, the test failed for load (over baseline) but it marks it as a success.  The opposite happens in the linked issue screenshot where the test was marked as a failure even though nothing was out of baseline.

### PR DESCRIPTION
**Pull request by yfangsl**
Fixes https://github.com/saucelabs/speedo/issues/28

We were treating jobResult.passed as true if it was undefined and false if it had anything in it.  This led to a reversed definition of "passed" which led to the issue when displaying results in analyzeReport.

This fix normalizes it so jobResult.passed = true if all results passed and false if any didn't.

### CODE REVIEW

File: src/commands/analyze.js [hunk 1]
```diff
@@ -123,13 +123,13 @@ export const handler = async (argv) => {
             })
         }
 
-        jobResult.passed = Object.values(jobResult.results).find((r) => !r.passed)
+        jobResult.passed = !Object.values(jobResult.results).find((r) => !r.passed)
```
Comment (adambabik) [inline src/commands/analyze.js]:
```suggestion
        jobResult.passed = Object.values(jobResult.results).find((r) => r.passed).length > 0
```

**Review by adambabik — APPROVED** [review]

File: src/commands/analyze.js (hunk 1 above)
Comment (yfangsl) [reply src/commands/analyze.js]: Shouldn't it just be Object.values(jobResult.results).find((r) => r.passed)?

I think find just returns the first element, not a list.

Comment (yfangsl) [reply src/commands/analyze.js]: Actually that doesn't seem right still, we'd still just pass if we had any results that passed when we should only pass if they all pass.  We want to pass only if we don't find non-passes.  I think this is why the double negation was needed.

Comment (adambabik) [reply src/commands/analyze.js]: Oh right. Sorry, I didn't notice that `find` is outside and the result is `undefined` or a single item. Confusing JS... So maybe `every()`?
```js
jobResult.passed = Object.values(jobResult.results).every((r) => r.passed)
```
or `some()` but it requires negation as well:
```js
jobResult.passed = !Object.values(jobResult.results).some((r) => !r.passed)
```

**Pull request merged**